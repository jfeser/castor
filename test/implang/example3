select([p_counter, c_counter],
  depjoin(
    ahashidx(dedup(select([id as p_id], log)),
    alist(select([counter, succ], filter(0.p_id = id && counter < succ, log)),
      atuple([ascalar(0.counter as p_counter), ascalar(0.succ as p_succ)], cross)),
    id_p),
  select([0.p_counter, c_counter],
  filter(c_id = id_c,
    aorderedidx(select([counter], log),
      alist(filter(counter = 0.counter, log),
        atuple([ascalar(0.id as c_id), ascalar(0.counter as c_counter)], cross)),
      0.p_counter, 0.p_succ)))))
