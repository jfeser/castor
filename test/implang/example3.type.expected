(FuncT
 (((FuncT
    (((HashIdxT
       ((IntT ((range (Interval 1 3))))
        (ListT
         ((TupleT
           (((IntT ((range (Interval 1 5)))) (IntT ((range (Interval 3 6)))))
            ((kind Cross))))
          ((count (Interval 1 2)))))
        ((key_count (Interval 3 3)))))
      (FuncT
       (((FuncT
          (((OrderedIdxT
             ((IntT ((range (Interval 1 5))))
              (ListT
               ((TupleT
                 (((IntT ((range (Interval 1 3))))
                   (IntT ((range (Interval 1 5)))))
                  ((kind Cross))))
                ((count (Interval 1 1)))))
              ((key_count (Interval 5 5))))))
           Child_sum)))
        (Width 2))))
     Child_sum)))
  (Width 2)))